### PR TITLE
Resolve `symfony/config:5.1` deprecation with `setDeprecated` method

### PR DESCRIPTION
**Pull request by andrew-demb**
inspired by https://github.com/doctrine/DoctrineBundle/commit/ca38c08c0b950733eb50587e4a1fda181e63b10d

#356

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -167,4 +168,27 @@ private function getOrganizeMigrationsModes() : array
...
+     * Returns the correct deprecation param's as an array for setDeprecated.
+     *
+     * Symfony/Config v5.1 introduces a deprecation notice when calling
+     * setDeprecation() with less than 3 args and the getDeprecation method was
+     * introduced at the same time. By checking if getDeprecation() exists,
+     * we can determine the correct param count to use when calling setDeprecated.
+     *
+     * @return string[]
+     */
+    private function getParamDeprecationMsg(string $message) : array
```
Comment (stof) [inline DependencyInjection/Configuration.php:182]: I would name it `getDeprecationParams` as it is about getting params

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -167,4 +168,27 @@ private function getOrganizeMigrationsModes() : array
 
         return $namesArray;
     }
+
+    /**
+     * Returns the correct deprecation param's as an array for setDeprecated.
```
Comment (greg0ire) [inline DependencyInjection/Configuration.php:173]: Either
```suggestion
     * Returns the correct deprecation params as an array for setDeprecated().
```

or

```suggestion
     * Returns the correct deprecation parameters as an array for setDeprecated().
```

File: DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -167,4 +168,27 @@ private function getOrganizeMigrationsModes() : array
...
+     *
+     * Symfony/Config v5.1 introduces a deprecation notice when calling
```
Comment (greg0ire) [inline DependencyInjection/Configuration.php:175]: Package names are not capitalized

```suggestion
     * symfony/config v5.1 introduces a deprecation notice when calling
```

File: DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -167,4 +168,27 @@ private function getOrganizeMigrationsModes() : array
...
+     * setDeprecation() with less than 3 args and the getDeprecation method was
```
Comment (greg0ire) [inline DependencyInjection/Configuration.php:176]:
```suggestion
     * setDeprecation() with less than 3 args and the getDeprecation() method was
```

File: DependencyInjection/Configuration.php [hunk 5]
```diff
@@ -167,4 +168,27 @@ private function getOrganizeMigrationsModes() : array
...
+     * introduced at the same time. By checking if getDeprecation() exists,
+     * we can determine the correct param count to use when calling setDeprecated.
```
Comment (greg0ire) [inline DependencyInjection/Configuration.php:178]: So is it `setDeprecation` or `setDeprecated`? Also, please add the missing parentheses

**Comment by goetas** [comment]
Thank you all!

**Comment by oleg-brizy** [comment]
Waiting for release

**Comment by MeroveeCharrue** [comment]
Hello,

Thank you for the fix :-)

I allow myself to reiterate the demand of @oleg-brizy :
Can this be shipped in a bugfix release soon ?

Thank you very much!

**Pull request merged**